myriad-scheduler/src/test/java/com/ebay/myriad/scheduler/fgs/YarnNodeCapacityManagerSpec.groovy

```diff
@@ -0,0 +1,118 @@
+package com.ebay.myriad.scheduler.fgs
+
+import com.ebay.myriad.scheduler.TaskFactory
+import com.ebay.myriad.scheduler.TaskUtils
+import com.ebay.myriad.scheduler.yarn.interceptor.InterceptorRegistry
+import com.ebay.myriad.state.NodeTask
+import com.ebay.myriad.state.SchedulerState
+import org.apache.hadoop.yarn.api.records.ContainerState
+import org.apache.hadoop.yarn.server.resourcemanager.scheduler.event.NodeResourceUpdateSchedulerEvent
+import org.apache.hadoop.yarn.util.resource.Resources
+import org.apache.mesos.Protos
+
+/**
+ *
+ * Tests for YarnNodeCapacityManager
+ *
+ */
+class YarnNodeCapacityManagerSpec extends FGSTestBaseSpec {
+
+  def "No Containers Allocated Due To Mesos Offers"() {
+    given:
+    def yarnNodeCapacityMgr = getYarnNodeCapacityManager()
+
+    def host = "test_host"
+    def slaveId = Protos.SlaveID.newBuilder().setValue(host + "_slave_id").build()
```

Comment:
a common groovy way to do this is:
`def slaveId = Protos.SlaveID.newBuilder().setValue("${host}_slave_id").build()`

however in this case yours may be more readable.